src/dataLoaders/components/collectorsWizard/TelegrafUIRefreshWizard.tsx

```diff
@@ -0,0 +1,178 @@
+// Libraries
+import React, {PureComponent} from 'react'
+import Loadable from 'react-loadable'
+import {connect, ConnectedProps} from 'react-redux'
+import {withRouter, RouteComponentProps} from 'react-router-dom'
+
+// Components
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import {Overlay} from '@influxdata/clockface'
+import {PluginCreateConfigurationFooter} from 'src/writeData/components/PluginCreateConfigurationFooter'
+
+const spinner = <div />
+const TelegrafUIRefreshStepSwitcher = Loadable({
+  loader: () =>
+    import(
+      'src/dataLoaders/components/collectorsWizard/TelegrafUIRefreshStepSwitcher'
+    ),
+  loading() {
+    return spinner
+  },
+})
+
+// Actions
+import {notify as notifyAction} from 'src/shared/actions/notifications'
+import {
+  clearSteps,
+  decrementCurrentStepIndex,
+  incrementCurrentStepIndex,
+  setBucketInfo,
+  setCurrentStepIndex,
+  setSubstepIndex,
+} from 'src/dataLoaders/actions/steps'
+
+import {clearDataLoaders} from 'src/dataLoaders/actions/dataLoaders'
+
+// Types
+import {AppState, Bucket, ResourceType} from 'src/types'
+import {PluginCreateConfigurationStepProps} from 'src/writeData/components/PluginCreateConfigurationWizard'
+
+// Selectors
+import {getAll} from 'src/resources/selectors'
+import {getOrg} from 'src/organizations/selectors'
+
+// Utils
+import {isSystemBucket} from 'src/buckets/constants'
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps & RouteComponentProps<{orgID: string}>
+
+@ErrorHandling
+class TelegrafUIRefreshWizard extends PureComponent<Props> {
+  public state = {pluginConfig: '', isValidConfiguration: false}
+
+  public componentDidMount() {
+    const {bucket, buckets} = this.props
+    if (!bucket && buckets && buckets.length) {
+      const {orgID, name, id} = buckets[0]
+      this.props.onSetBucketInfo(orgID, name, id)
+    }
+    this.props.onSetCurrentStepIndex(0)
+  }
+
+  public render() {
+    const {currentStepIndex} = this.props
+    let overlayBodyClassName = 'data-loading--overlay'
+
+    if (currentStepIndex === 1) {
+      overlayBodyClassName = ''
+    }
+
+    return (
+      <Overlay visible={true}>
```

Comment:
so is this overlay always going to be shown? will the component always be in the DOM?

Reply:
Yes, technically it is always shown once mounted. The `onDismiss` handler uses `history.push` to re-route the user which causes the Overlay to unmount from the DOM.

But this is a good observation. Clearly we are only relying on history.push to unmount the Overlay. I think it would be better to also set the visibility to `false` in the `onDimiss` handler. I will push up that change.